### DISCUSSION TITLE
Unify the position of numbering

### DISCUSSION THREAD

**Discussion by GrimPixel** [IndiePython/nodezator#64] [Feature Request]
In those “Builtin picks”, “Standard lib picks”, “Useful encapsulations”, the numbering is at the lower center of the node; in “Common operations”, they are at the upper-left corner; in “New data node”, they are at the upper-left corner with a colon and two spaces to separate it from the name.
I think the lower center is good, like the numbering in documents.

**Comment by KennedyRichard** [comment]
Admittedly, adopting the same positioning is a fair expectation, but the positioning is adopted in order to make these nodes a bit more compact.

For instance, in data nodes, by putting the numbering to the side, I just need to increase the size of the label horizontally by the number of characters used and the spaces + colon. If I were to put the numbering below the title of the node, I'd have to double the height of the node just to put the numbering below it. See the difference in size below:

![data_node_label_comparison](https://github.com/IndiePython/nodezator/assets/18728298/04e67b9a-c560-4334-aa84-770a3743e37e)

The node doubles in height just for a little extra information. However, in the case of operation nodes it shouldn't be a problem because it would increase the used are below but free the used are above the node, so I think the change would indeed be appropriate.

For both data nodes and operation nodes though, I think an even better improvement would be to style the numbering the same way it is styled in the other nodes (with a black background). I think this way the numbering would be more readable and more easy to distinguish from the other elements in the node.

This would be more or less the intended result:

![styled_id_number](https://github.com/IndiePython/nodezator/assets/18728298/893b8275-4b20-49cd-aa42-d0c3e5e12d0d)

If I kept the numbering of the operation node on the topleft corner it would still be quite readable with the black background style though:

![styled_id_number_v2](https://github.com/IndiePython/nodezator/assets/18728298/93f56e1c-a5f9-4039-868a-0598cfe5b737)

I'll keep your suggestion in mind, but I'm more inclined to keep the positioning as it is and only add the black background. I think this way they will be readable enough and still compact. I also think that for operation nodes, positioning the number at the topleft corner is a bit more pleasing, but that's just my personal preference. When I change this in the future I'll ask other people in the community what they think would be better, so we have more perspectives on the matter.
